### ISSUE TITLE
Is source code of CPN-AMI available ?

### ISSUE THREAD

**Issue by gl-yziquel** [lip6/Coloane#48]
Is source code of CPN-AMI available ?

**Comment by yanntm** [comment]
I don't believe it's on github, but yes it should still exist on some repo at lip6. You are interested in some components ?

**Comment by gl-yziquel** [comment]
@yanntm I'm pretty much curious to have a look at the codebase. I'm dabbling in so-called "production systems" of the 1970s, and currently the SOAR kernel, and I wonder how much petri nets are amenable to model stuff happening in these stuff. For now, I'm more into an investigative phase.

**Comment by yanntm** [comment]
CPN-AMI was mostly a collection of analysis tools, with a server protocol so you could query the tools from a distant machine. The tools themselves had diverse origins, e.g. GreatSPN from Torino Univ. in Italy, PROD from Helsinki, Lola from Germany... Some of these tools are still active, for instance GreatSPN, and now have their own front-end.
Overall CPN-AMI was mostly a driver for back end (model-checking) tools, handling format conversions and parse of results.
And large parts were written in Ada, notably the server component (back end tools mostly C at the time).

I'm reluctant to spend the time searching for the CPN-AMI source and extracting/converting to git, making sure the licenses are ok for me to publish etc... unless you have a more pressing need for the code base. From my POV it's mostly obsolete code, you should rather use some tools that are maintained to see what one can do with Petri nets e.g. Tapaal, ITS-Tools, GreatSPN...

**Comment by yanntm** [comment]
Or (after reading a bit on SOAR) you want to browse the source code to have a look at a historical System-of-Systems that was already operational in the 90's ? CPN-AMI was an integration platform, I'm not sure it deserves to be called a SoS, it's more a set of tools with adapters to a kind of software bus, so more in the spirit of a kind of middleware than a SoS with various peers collaborating.

**Comment by gl-yziquel** [comment]
@yanntm do not bother too much then. I do have an archeological bent towards computer stuff. So I'd love it. But if it's a licensing headache, do not bother.

Thank you anyway.
